tests/test_project.py

```diff
@@ -2511,6 +2523,150 @@ def test_raise_exception_in_hook_cmd(self):
         assert "RuntimeError" in error_output
 
 
+class TestHooksTrackOperations(TestHooksSetUp):
+    project_class = define_hooks_track_operations_project._HooksTrackOperations
+    entrypoint = dict(
+        path=os.path.realpath(
+            os.path.join(
+                os.path.dirname(__file__), "define_hooks_track_operations_project.py"
+            )
+        )
+    )
+    log_fname = ".operations_log.txt"
+
+    @pytest.fixture(
+        params=[
+            (
+                "strict_git_false",
+                define_hooks_track_operations_project.HOOKS_ERROR_MESSAGE,
+            ),
+            ("strict_git_false_cmd", "non-zero exit status 42"),
+        ]
+    )
+    def strict_git_false_operation_info(self, request):
+        return request.param
+
+    @pytest.fixture(
+        params=[
+            (
+                "strict_git_true",
+                define_hooks_track_operations_project.HOOKS_ERROR_MESSAGE,
+            ),
+            ("strict_git_true_cmd", "non-zero exit status 42"),
+        ]
+    )
+    def strict_git_true_operation_info(self, request):
+        return request.param
+
+    def split_log(self, job):
+        with open(job.fn(self.log_fname)) as f:
+            values = f.read().split("\n")
+        if values[-1] == "":
+            values = values[:-1]
+        return values
+
+    def test_metadata(self, project, job, strict_git_false_operation_info):
+        operation_name, error_message = strict_git_false_operation_info
+        assert not job.isfile(self.log_fname)
+
+        time = datetime.datetime.now(datetime.timezone.utc)
+        if job.sp.raise_exception:
+            with pytest.raises(subprocess.CalledProcessError):
+                self.call_subcmd(f"run -o {operation_name} -j {job.id}")
+        else:
+            self.call_subcmd(f"run -o {operation_name} -j {job.id}")
+
+        assert job.isfile(self.log_fname)
+        values = self.split_log(job)
+
+        # There will always be 4 entries in this file because
+        # for every operation, we install the following hooks
+        # at the operation and project level:
+        # (1) on_start and (2) on_exception / on_success
+        assert len(values) == 4
+        for value in values[:2]:
+            metadata = json.loads(value)
+            assert metadata["stage"] == "prior"
+            # I think job._project.path gives us a relative path in the test, while
+            # job._project.path in hooks.utils.collect_metadata gives us the absolute path
+            # I don't know why this is happening.
+            assert job._project.path in metadata["project"]["path"]
+            assert metadata["project"]["schema_version"] == job._project.config.get(
+                "schema_version"
+            )
+            # Just assumed that the operation should start within 5 minutes
+            # from where we recorded the time line 2477. This seems generous and
```

Comment:
This line number seems to be outdated. Can we refer to that place some other way?

Reply:
Repeats below.

Reply:
I removed the line number reference. Would that work?